tests/jose/algorithms/signing/secp256k1.spec.ts

```diff
@@ -1,31 +1,93 @@
+      expect(signatureBytes.length).to.equal(64); // DER format would be 70 bytes
+    });
+  });
+
+  describe('deriveChildPublic/PrivateKey()', () => {
+    it('should derive the same key pair counterparts when given the same tweak input', async () => {
+      const { publicKey, privateKey } = await Secp256k1.generateKeyPairRaw();
+
+      const tweakInput = TestDataGenerator.randomBytes(32);
+      const derivedPrivateKey = Secp256k1.deriveChildPrivateKey(privateKey, tweakInput);
+      const derivedPublicKey = Secp256k1.deriveChildPublicKey(publicKey, tweakInput);
+
+      const publicKeyFromDerivedPrivateKey = await Secp256k1.getPublicKey(derivedPrivateKey);
+      expect(Comparer.byteArraysEqual(derivedPublicKey, publicKeyFromDerivedPrivateKey)).to.be.true;
+    });
+  });
+
+  describe('derivePublic/PrivateKey()', () => {
+    it('should be able to derive same key using different ancestor along the chain path', async () => {
+      const { publicKey, privateKey } = await Secp256k1.generateKeyPairRaw();
+
+      const fullPathToG = 'a/b/c/d/e/f/g';
+      const fullPathToD = 'a/b/c/d';
+      const relativePathFromDToG = 'e/f/g';
+
+      // testing public key derivation from different ancestor in the same chain
+      const publicKeyG = await Secp256k1.derivePublicKey(publicKey, fullPathToG);
+      const publicKeyD = await Secp256k1.derivePublicKey(publicKey, fullPathToD);
+      const publicKeyGFromD = await Secp256k1.derivePublicKey(publicKeyD, relativePathFromDToG);
+      expect(Comparer.byteArraysEqual(publicKeyG, publicKeyGFromD)).to.be.true;
+
+      // testing private key derivation from different ancestor in the same chain
+      const privateKeyG = await Secp256k1.derivePrivateKey(privateKey, fullPathToG);
+      const privateKeyD = await Secp256k1.derivePrivateKey(privateKey, fullPathToD);
+      const privateKeyGFromD = await Secp256k1.derivePrivateKey(privateKeyD, relativePathFromDToG);
+      expect(Comparer.byteArraysEqual(privateKeyG, privateKeyGFromD)).to.be.true;
+
+      // testing that the derived private key matches up with the derived public key
+      const publicKeyGFromPrivateKeyG = await Secp256k1.getPublicKey(privateKeyG);
+      expect(Comparer.byteArraysEqual(publicKeyGFromPrivateKeyG, publicKeyG)).to.be.true;
+    });
+
+    it('should derive the same public key using either the private or public counterpart of the same key pair', async () => {
+      const { publicKey, privateKey } = await Secp256k1.generateKeyPairRaw();
+
+      const path = '1/2/3/4/5/6/7/8/9/10';
+
+      const derivedKeyFromPublicKey = await Secp256k1.derivePublicKey(publicKey, path);
+      const derivedKeyFromPrivateKey = await Secp256k1.derivePublicKey(privateKey, path);
+      expect(Comparer.byteArraysEqual(derivedKeyFromPublicKey, derivedKeyFromPrivateKey)).to.be.true;
+    });
+
+    it('should derive the same public key using either the private or public counterpart of the same key pair', async () => {
```

Comment:
Related to my other comment, should we also test derivation path validation for paths that include like emojis and empty strings?

Reply:
resolving based on your reply above. ty